utils/cloneObject.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Creates a deep clone of the provided object.
+ *
+ * This function serializes the object to a JSON string and then parses it back to a new object,
+ * effectively creating a deep copy. Note that this method may not handle all edge cases, such as
+ * objects with circular references, functions, or special object types like `Date` or `Map`.
+ *
+ * @typeParam T - The type of the object to be cloned.
+ * @param obj - The object to be cloned.
+ * @returns A deep clone of the provided object.
+ */
+export const cloneObject = <T>(obj: T): T => {
+  return JSON.parse(JSON.stringify(obj))
```

Comment:
Yeah this works well with primitives but as you mention it will not work well with symbols, date, map, functions etc.. thats why lodash implements it on a different way: https://github.com/lodash/lodash/blob/f299b52f39486275a9e6483b60a410e06520c538/lodash.js#L2662

Reply:
Yes, for current Grid use it's sufficient. I felt lodash is an overkill here.

Reply:
yeah I assumed it was the case